app/src/navigation/MainTabNavigator.tsx

```diff
@@ -1,45 +1,92 @@
-import React from "react";
+import React, { useEffect } from "react";
 import { createBottomTabNavigator } from "@react-navigation/bottom-tabs";
 import { FontAwesome5 } from "@expo/vector-icons";
 import DocumentsStackNavigator from "./DocumentsStackNavigator";
 import ProfileStackNavigator from "./ProfileStackNavigator";
 import VendorsStackNavigator from "./VendorsStackNavigator";
-import { ParamListBase, RouteProp } from "@react-navigation/native";
+import {
+  CommonActions,
+  ParamListBase,
+  RouteProp,
+  useNavigation
+} from "@react-navigation/native";
 import colors from "../styles/colors";
+import { MainTabNavigation } from "../types/navigation";
+import { Linking } from "react-native";
 
 export type MainTabParamList = {
   Profile: undefined;
-  SupportedExchanges: undefined;
+  SupportedExchanges: undefined | { screen: string; params: any };
   DocumentsTab: undefined;
 };
 
 const Tab = createBottomTabNavigator<MainTabParamList>();
 
-const TabNavigator = () => (
-  <Tab.Navigator
-    initialRouteName="Profile"
-    screenOptions={({ route }) => ({
-      headerShown: false,
-      tabBarIcon: (tabProps) => renderTabIcon(route, tabProps),
-      tabBarActiveTintColor: colors.blue,
-      tabBarInactiveTintColor: colors.gray
-    })}
-  >
-    <Tab.Screen name="Profile" component={ProfileStackNavigator} />
-    <Tab.Screen
-      name="SupportedExchanges"
-      component={VendorsStackNavigator}
-      options={{ title: "Supported Exchanges" }}
-    />
-    <Tab.Screen
-      name="DocumentsTab"
-      options={{
-        title: "Documents"
-      }}
-      component={DocumentsStackNavigator}
-    />
-  </Tab.Navigator>
-);
+type Navigation = MainTabNavigation<"Profile">;
+
+const TabNavigator = () => {
+  const navigation = useNavigation<Navigation>();
+
+  useEffect(() => {
+    Linking.addEventListener("url", ({ url }) => {
+      if (url.includes("exchange/vendors")) {
```

Comment:
To prevent indent hell is it worth changing this to a guard clause?

Reply:
react navigation quirk

Reply:
react navigation quirk